content/blog/2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,273 @@
+  machine types which are roughly the same for all supported clouds. For
+  example,
+  [l+k80](https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#l+k80)
+  stands for a "Large, with (at least) 12 CPU cores, 112 GB of RAM and 2 GPU
+  devices".
+- _workdir_: Specify a directory on your local machine relative to your project
+  folder which you would like to sync with the remote machine. This way you can
+  share your whole project or parts of it with a remote machine. In my example,
+  I am sharing the whole project.
+- _script_ (**required)**: This is where the Iterative Provider's magic happens,
+  i.e. this is where we define a script that should be run on a provisioned
+  machine.
+
+Take a look at the script. In the simplest scenario, all we need to do on a new
+machine to run the training script is to set up the Python environment with
+required libraries. If you simply want to train your model on a machine with
+more memory, this may be enough. However, if you want your training code to
+leverage GPUs, the script will look a bit different.
+
+## Training with GPU
+
+There are several ways you can leverage GPU devices on a remote machine. You can
+install all the required drivers and dependencies "manually" via a script, you
+can use an existing Docker image or build your own, or, in the case of AWS, you
+can take advantage of an
+[existing pre-configured Deep Learning AMI](https://docs.aws.amazon.com/dlami/latest/devguide/options.html).
+In this tutorial, we'll use an AMI, as it is the quickest way to start using
+GPUs on an AWS EC2 machine.
+
+```hcl
+resource "iterative_task" "tpi-examples-gpu" {
+    name      = "ami-gpu-example"
+    cloud     = "aws"
+    region    = "us-east-2"
+    machine   = "l+k80"
+    disk_size = "130"
+    image     = "ubuntu@898082745236:x86_64:Deep Learning AMI (Ubuntu 18.04) Version 54.0"
+
+    workdir { input = "." }
+    script = <<-END
+    #!/bin/bash
+    pip3 install -r requirements.txt
+    python3 src/train.py
+    END
+}
+```
+
+To use a Deep Learning AMI, you need to configure it via the `image` argument
+that has the following structure: `{user}@{owner}:{architecture}:{name}`. You
+can learn more about AMIs in the official
+[AWS documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html).
+
+The Deep Learning image that you choose may require larger disk size than the
+provided default, so you may need to increase it. The image I picked (see
+screenshot above) expectes volume of at least 130 GB, and I can configure this
+in the `disk_size` argument.
+
+You may also notice that my script got a bit shorter. This is because I no
+longer need to install Python, and only need to make sure to have my
+requirements met.
+
+## Ready… Set… Apply!
+
+Whether you want to go with the basic example, or the GPU-enabled training, you
+can run `terraform plan` to review what steps Terraform is going to take to make
```

Comment:
For this last section, do we need more code blocks instead of just mentioning the `terraform` commands? The blog design currently doesn't highlight them that much, they can get lost in the regular text.

![image](https://user-images.githubusercontent.com/1477535/152416276-6862b847-8931-46f0-9e1a-549c0d106fe5.png)

Not sure what to suggest to address this. Maybe it's OK since you're not writing about Terraform.

Reply:
made into separate code blocks